### ISSUE TITLE
Errors thrown from validation rules are hidden from user

### ISSUE THREAD

**Issue by jcheng5** [rstudio/shinyvalidate#18]
Currently, a rule that throws results in the input passing validation (at least that's what it looks like to me on cursory investigation). In the below example, the text input has no visual indicator that anything is wrong, and `iv$is_valid()` returns `TRUE`.

An error in validation should probably manifest as a validation failure. We need to decide what kind of message to display:

1. "An unexpected error occurred while processing this input"
2. Actually display the message (though when Shiny [error sanitization](https://shiny.rstudio.com/articles/sanitize-errors.html) is enabled, we'd need to fall back to a generic message like option 1 for most errors)

```r
library(shiny)
library(shinyvalidate)

ui <- fluidPage(
  textInput("input1", NULL),
  p(
    "Validation passing: ",
    textOutput("validation_result", container = tags$strong)
  )
)

server <- function(input, output, session) {
  iv <- InputValidator$new()
  iv$add_rule("input1", ~ stop("boom"))
  iv$enable()
  
  output$validation_result <- renderText({
    if (iv$is_valid()) "yes" else "no"
  })
}

shinyApp(ui, server)
```
